src/main/scala-2.12/org/apache/james/gatling/jmap/scenari/JmapInboxHomeLoadingScenario.scala

```diff
@@ -0,0 +1,56 @@
+package org.apache.james.gatling.jmap.scenari
+
+import io.gatling.core.Predef._
+import io.gatling.core.feeder.FeederBuilder
+import io.gatling.core.structure.ScenarioBuilder
+import io.gatling.http.Predef._
+import io.gatling.http.request.builder.HttpRequestBuilder
+import org.apache.james.gatling.jmap._
+
+class JmapInboxHomeLoadingScenario {
+
+  object Keys {
+
+    val inbox = "inboxID"
+
+    val messageIds = "messageIds"
+
+    val messagesDetailList = "messagesDetailList"
+
+  }
+
+
+  private val mailboxListPath = "[0][1].list"
+  private val inboxIdPath = s"$$$mailboxListPath[?(@.role == 'inbox')].id"
+
+  val listAllMailboxesAndSelectFirstOne: HttpRequestBuilder =
+    JmapAuthentication.authenticatedQuery("getMailboxes", "/jmap")
+      .body(StringBody("""[["getMailboxes",{},"#0"]]"""))
+      .check(jsonPath(inboxIdPath).find.saveAs(Keys.inbox))
+
+  val getVacationResponse: HttpRequestBuilder =
+    JmapAuthentication.authenticatedQuery("getVacationResponse", "/jmap")
+      .body(StringBody(s"""[["getVacationResponse",{},"#0"]]"""))
+
+  val listMessagesInInbox: HttpRequestBuilder =
+    JmapAuthentication.authenticatedQuery("listMessagesInInbox", "/jmap")
+      .body(StringBody("""[["getMessageList",{"filter":{"inMailboxes":["${inboxID}"],"text":null},"sort":["date desc"],"collapseThreads":false,"fetchMessages":false,"position":0,"limit":200,"#0"]]"""))
+      .check(jsonPath("$[0][1].messageIds[*]").findAll.saveAs(Keys.messageIds))
+
+
+  private val getMessages: HttpRequestBuilder =     JmapAuthentication.authenticatedQuery("getMessages", "/jmap")
+    .body(StringBody("""[["getMessages",{"properties": ["id","blobId","threadId","headers","subject","from","to","cc","bcc","replyTo","preview","date","isUnread","isFlagged","isDraft","hasAttachment","mailboxIds","isAnswered","isForwarded"],"ids": ${messageIds},"#0"]]"""))
+
+  def generate(feederBuilder: FeederBuilder): ScenarioBuilder = {
+
+
+    scenario("JmapHomeLoadingScenario")
+      .feed(feederBuilder)
+      .exec(CommonSteps.authentication())
+      .exec(RetryAuthentication.execWithRetryAuthentication(listAllMailboxesAndSelectFirstOne, JmapMailboxes.getMailboxesChecks))
+      .exec(RetryAuthentication.execWithRetryAuthentication(getVacationResponse, Nil))
```

Comment:
could we make `execWithRetryAuthentication` more developer friendly and make the second argument optional ?

Reply:
I think there is no need for this refactor. Since i removed the getVacationResponse call.
We check all the response with at least the status code now